cumulusci/core/tests/test_github.py

```diff
@@ -101,11 +129,91 @@ def test_get_github_api_for_repo__not_installed(self, create_token):
         with mock.patch.dict(
             os.environ, {"GITHUB_APP_KEY": "bogus", "GITHUB_APP_ID": "1234"}
         ):
-            with self.assertRaises(GithubException):
+            with pytest.raises(GithubException):
                 get_github_api_for_repo(None, "TestOwner", "TestRepo")
 
     @responses.activate
     def test_validate_service(self):
         responses.add("GET", "https://api.github.com/rate_limit", status=401)
-        with self.assertRaises(GithubException):
+        with pytest.raises(GithubException):
             validate_service({"username": "BOGUS", "password": "BOGUS"})
+
+    @responses.activate
+    def test_get_pull_request_by_branch_name(self, mock_util, repo):
+        self.init_github()
+        mock_util.mock_pulls(
+            pulls=self.get_expected_pull_requests(1),
+            head=repo.owner.login + ":" + "some-other-branch",
+        )
+        pull_request = get_pull_request_by_branch_name(repo, "some-other-branch")
+        assert pull_request is not None
+
+        # ConnectionError present when we reachout with
+        # a branch name (url parameter) that we aren't expecting
+        with pytest.raises(ConnectionError):
+            get_pull_request_by_branch_name(repo, "does-not-exist")
```

Comment:
@davisagli - I think my comment can be improved here. In this case the `ConnectionError` is coming from the `responses` module. In this case, we add a response that contains the `head=` url parameter. The first time we call `get_pull_request_by_branch_name` the given branch name matches the expected parameter value.

In the next case we use a value for `head` that doesn't exist and so the endpoint doesn't fully match the response that we added so it fails with a `ConnectionError`. 

[The docs](https://github.com/getsentry/responses) state, 'If you attempt to fetch a url which doesn't hit a match, responses will raise a ConnectionError'

Reply:
Oh that makes more sense -- but then I think you want to import ConnectionError from requests.exceptions instead of github3.exceptions? (The latter must coincidentally import the former, since the test works. But it was the github3 import that confused me about where the ConnectionError comes from.)

Reply:
ahh ok, didn't even think of that. I'll switch the import and rerun tests to ensure all is well.

Reply:
Oh I see, github3 is catching the requests ConnectionError and reraising it as its own exception class. (in models.py in github3)